### PR TITLE
add reading input from file (add file close)

### PR DESCRIPTION
**Pull request by pstanisz**
This PR is just for review purposes.

### CODE REVIEW

**Review by pstanisz — COMMENTED** [review]
For:
#define VOWELS  "aeiou"

It is a "C" approach, correct but you can experiment with constexpr and modern C++ approach.

**Review by pstanisz — COMMENTED** [review]
Code:
```
bool    is_vowels(char character) {
    char    lower;
    lower = tolower(character);
    for (int i = 0; VOWELS[i] != 0; i++) {
        if (lower == VOWELS[i])
            return true;
    }
    return false;
}
```
Modern C++ approach:
`auto lower = std::tolower(character);`
No need to declare a variable at the beginning of function, this was a "C" approach.

Using prefix incrementation/decrementation is always a good practice in loops: `++i;`

**Review by pstanisz — COMMENTED** [review]
`void count_words(char character, int *vowels, int *consonants, int *others)`
It would be better to use references in C++:
`void count_words(char character, int &vowels, int &consonants, int &others)`

Another approach is to use a tuple and return all of the results within a tuple.

**Comment by mjtieto** [comment]
`int     read_from_file(char *file_name)`
I'd take the name by a `const std::string&`: `int     read_from_file(const std::string &file_name)`

**Comment by mjtieto** [comment]
```cpp
    std::ifstream   input_file;
    input_file.open(file_name);
```
You could use constructor directly: `std::ifstream input_file{ file_name };`

**Comment by mjtieto** [comment]
`int     words_checker(std::istream &read_file, char const *stop)`
`stop` could be a `const std::string&` and have a default value:
`int     words_checker(std::istream &read_file, const std::string& stop = {})`

**Comment by dburtnja** [comment]
> For:
> #define VOWELS "aeiou"
> 
> It is a "C" approach, correct but you can experiment with constexpr and modern C++ approach.

File: exercise_0/main.cpp [hunk 1]
```diff
@@ -59,6 +59,7 @@ int     read_from_file(char *file_name) {
         return 1;
     }
     words_checker(input_file, nullptr);
+    input_file.close();
```
Comment (mjtieto) [inline exercise_0/main.cpp:58]: This is actually not needed. File will be closed in its destructor, anyway.

**Pull request closed without merging**